pandas/io/common.py

```diff
@@ -165,11 +165,80 @@ def get_filepath_or_buffer(filepath_or_buffer, encoding=None,
         except boto.exception.NoAuthHandlerFound:
             conn = boto.connect_s3(anon=True)
 
```

Comment:
out this class outside the function
it's private anyhow as long as u don't export thru the API